agent/src/com/jonnyzzz/teamcity/virtual/VMDetector.java

```diff
@@ -61,26 +61,30 @@ private void detectVagrant(@NotNull final BuildAgentConfiguration config) {
       }
 
       private void detectDocker(@NotNull final BuildAgentConfiguration config) {
-        if (!SystemInfo.isLinux) {
-          LOG.debug("Docker is only available under Linux");
-          return;
-        }
 
-        final String output = executeCommandWithShell("docker", "docker --version");
+        final String output = executeCommandWithShell("docker", "docker version");
         if (output == null) return;
 
-        String ver = output.toLowerCase().trim();
-        ver = ver.replaceAll("\\s*docker\\s+version\\s+", "");
-        ver = ver.replaceAll(",?\\s+build\\s+", "-");
+        final String dockerVersion = executeCommandWithShell("docker", "docker version --format {{.Server.Version}}");
```

Comment:
`docker version --format`` if relatively new key. I did experiment and it showed Docker 1.7.1 does not support this option. Currently, it would mean we no longer support older docker. 
Do we really need this limitation?